finesse/hardware/plugins/spectrometer/ftsw500_interface.py

```diff
@@ -70,135 +60,141 @@ def __init__(self) -> None:
         """Create a new FTSW500Interface."""
         super().__init__()
 
-        self._requester = socket(AF_INET, SOCK_STREAM)
-        self._requester.settimeout(FTSW500_TIMEOUT)
-
-        self._requester.connect((FTSW500_HOST, FTSW500_PORT))
-
-        self._status = SpectrometerStatus.UNDEFINED
+        sock = socket(AF_INET, SOCK_STREAM)
+        sock.settimeout(FTSW500_TIMEOUT)
+        sock.connect((FTSW500_HOST, FTSW500_PORT))
+        self._socket = sock
 
+        # Timer to poll status
         self._status_timer = QTimer()
-        self._status_timer.timeout.connect(self._request_status)
+        self._status_timer.timeout.connect(self.pubsub_errors(self._update_status))
         self._status_timer.setInterval(int(FTSW500_POLLING_INTERVAL * 1000))
         self._status_timer.setSingleShot(True)
 
-        self._request_status()
-
-    def _set_output_path_prefix(self, output_path_prefix: str) -> None:
-        """Set the output path prefix for FTSW500 data."""
-        self._requester.sendall(
-            f"setOutputPathPrefixField&{output_path_prefix}\n".encode()
-        )
-        self._requester.recv(1024)
-
-    def _set_output_prefix(self, output_prefix: str) -> None:
-        """Set the output prefix for FTSW500 data."""
-        self._requester.sendall(f"setOutputPrefixOnlyField&{output_prefix}\n".encode())
-        self._requester.recv(1024)
-
-    def set_data_description(self, data_description: str) -> None:
-        """Set the description for FTSW500 data."""
-        self._requester.sendall(
-            f"setDataDescriptionField&{data_description}\n".encode()
-        )
-        self._requester.recv(1024)
-
-    def set_num_measurements(self, num_measurements: int) -> None:
-        """Set the number of measurements for FTSW500 to record."""
-        self._requester.sendall(
-            f"setNumberMeasurementField&{num_measurements}\n".encode()
-        )
-        self._requester.recv(1024)
-
-    def set_temperature(self, temperature: float) -> None:
-        """Set the temperature for FTSW500 data."""
-        self._requester.sendall(f"setTemperatureField&{temperature}\n".encode())
-        self._requester.recv(1024)
```

Comment:
While this is currently unused, Jon mentioned it would be useful to be able to send the black-body temperatures to FTSW500 to compute spectra. Exactly how and how often we do this isn't clear, so we should discuss this with Jon.

Reply:
As above, I thought we could wait till we needed it.